### PR TITLE
TensorFlow estimator integration example with only v2 APIs

### PR DESCRIPTION
**Pull request by nuka137**
This PR fixes TensorFlow estimator integration example to use v2 APIs. (c.f. #924 )
This PR also fixes name of variables which mentioned in [here](https://github.com/optuna/optuna/pull/924#pullrequestreview-358708888).

Related Issue: #930

### CODE REVIEW

File: examples/pruning/tensorflow_estimator_integration.py [hunk 1]
```diff
@@ -24,90 +24,78 @@
...
-        x={"x": train_data}, y=train_labels, batch_size=BATCH_SIZE, num_epochs=None, shuffle=True)
-
     train_spec = tf.estimator.TrainSpec(
         input_fn=train_input_fn, max_steps=TRAIN_STEPS, hooks=[optuna_pruning_hook])
 
-    eval_input_fn = tf.compat.v1.estimator.inputs.numpy_input_fn(
-        x={"x": eval_data}, y=eval_labels, num_epochs=1, shuffle=False)
-
     eval_spec = tf.estimator.EvalSpec(
         input_fn=eval_input_fn, steps=EVAL_STEPS, start_delay_secs=0, throttle_secs=0)
```
Comment (toshihikoyanase) [inline examples/pruning/tensorflow_estimator_integration.py]: Please remove `step=EVAL_STEPS` similarly to https://github.com/optuna/optuna/pull/924#discussion_r378791734. If `step` is set, we restrict the number of examples using two constants (i.e., `EVAL_STEPS` and `N_TEST_BATCHES`) and it seems to be a bit complicated. 
(c.f., https://www.tensorflow.org/api_docs/python/tf/estimator/EvalSpec#args).

Comment (nuka137) [reply examples/pruning/tensorflow_estimator_integration.py]: Thanks, I fixed it.

File: examples/pruning/tensorflow_estimator_integration.py [hunk 2]
```diff
@@ -24,90 +24,78 @@
 TRAIN_STEPS = 1000
 EVAL_STEPS = 100
```
Comment (toshihikoyanase) [inline examples/pruning/tensorflow_estimator_integration.py]: Please remove this line because `EVAL_STEPS` is not used anymore.
```suggestion
```

Comment (nuka137) [reply examples/pruning/tensorflow_estimator_integration.py]: Sorry, I missed it.
I removed `EVAL_STEPS`.

**Comment by toshihikoyanase** [comment]
@nuka137 
Thank you for your updates! It looks good to me.
Please merge master into this branch to resolve CI failures.

**Pull request merged**